core/include/vc/core/types/Volume.hpp

```diff
@@ -222,10 +233,14 @@ class Volume : public DiskBasedObjectBaseClass,
     /** Slice cache */
     mutable SliceCache::Pointer cache_{DefaultCache::New(DEFAULT_CAPACITY)};
     /** Cache mutex for thread-safe access */
-    mutable std::mutex cacheMutex_;
+    mutable std::shared_mutex cacheMutex_;
+    /** Per-slice mutexes */
+    mutable std::vector<std::mutex> sliceMutexes_;
```

Comment:
Initializing a mutex for every slice feels wasteful, especially when we know we're caching slices.